### PR TITLE
Fix GKE integration with clusterupstreamrefresher

### PR DESCRIPTION
**Pull request by mjura**
Add missing gke-refresh setting and missing bits for GKE
cloud provider in clusterupstreamrefresher.

### CODE REVIEW

**Review by cmurphy — DISMISSED** [review]
Thanks for catching this

**Review by thedadams — DISMISSED** [review]

File: pkg/controllers/management/clusterupstreamrefresher/cluster_upstream_refresher.go [hunk 1]
```diff
@@ -120,16 +121,14 @@ func getProviderAndReadyStatus(cluster *mgmtv3.Cluster) (string, bool) {
...
-	case apimgmtv3.ClusterDriverGKE:
-		refreshInterval = settings.GKEUpstreamRefresh.GetInt()
-	default:
-		return 300, nil
+	providerRefreshInterval := settings.GetSettingByID(fmt.Sprintf(refreshSettingFormat, provider))
+	if providerRefreshInterval == "" {
+		return 300, fmt.Errorf("could not get refresh interval from setting")
+	}
+    refreshInterval, err := strconv.Atoi(providerRefreshInterval)
+	if providerRefreshInterval == "" {
```
Comment (rmweir) [inline pkg/controllers/management/clusterupstreamrefresher/cluster_upstream_refresher.go:130]: I think this is an unnecessary indent.

**Review by rmweir — CHANGES_REQUESTED** [review]

**Review by thedadams — DISMISSED** [review]

**Review by rmweir — COMMENTED** [review]
lint issues, otherwise LGTM

File: pkg/controllers/management/clusterupstreamrefresher/cluster_upstream_refresher.go (hunk 1 above)
Comment (mjura) [reply pkg/controllers/management/clusterupstreamrefresher/cluster_upstream_refresher.go]: sorry, I used `go fmt` again

**Review by thedadams — APPROVED** [review]

**Review by rmweir — APPROVED** [review]
LGTM

**Review by cbron — APPROVED** [review]
Lgtm. Only weirdness may be if provider isn't getting used correctly but it should be fine.

**Comment by cbron** [comment]
backfix for: https://github.com/rancher/rancher/issues/31011

**Pull request merged**